pyxem/signals/symmetry1d.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2020 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Signal class for pearson correlation and symmetry coefficient."""
+
+from hyperspy.signals import Signal2D, BaseSignal, Signal1D
+#from hyperspy._signals.lazy import LazySignal
+import numpy as np
+
+
+class Symmetry1D(Signal1D):
+    _signal_type = "symmetry"
+
+    def __init__(self, *args, **kwargs):
+        """Create a Symmetry object from a numpy.ndarray.
+
+        Parameters
+        ----------
+        *args :
+            Passed to the __init__ of Signal1D. The first arg should be
+            a numpy.ndarray
+        **kwargs :
+            Passed to the __init__ of Signal1D
+        """
+        super().__init__(*args, **kwargs)
+
+        self.decomposition.__func__.__doc__ = BaseSignal.decomposition.__doc__
+
+    def get_symmetry_coefficient(self):
+        """Return symmetry coefficient from pearson correlation function at all real
+        space positions (n from 2 to 10).
+
+        Returns
+        -------
+        sn: Signal1D
+            Symmetry coefficient
+        """
```

Comment:
Can you add a references to the paper you've mentioned in the PR here please.

Reply:
This method here is actually a bit different from what they did in the paper I mentioned. The correlation part is similar but they didn't incorporate symmetry coefficient, so I'm not sure if it would be accurate to reference that paper. We are writing a new paper based on this specific method currently. Maybe we can reference to that one in the future.

Reply:
Yup, happy with that.